File: placement-flutter/storyly_placement_flutter/android/src/main/kotlin/com/appsamurai/storyly/storyly_placement_flutter/common/data/widgets/SwipeCardData.kt
package com.appsamurai.storyly.storyly_placement_flutter.common.data.widgets

import com.appsamurai.storyly.core.STRWidgetType
import com.appsamurai.storyly.swipecard.data.SwipeCardDataPayload
import com.appsamurai.storyly.swipecard.data.model.SwipeCardPayload
import com.appsamurai.storyly.swipecard.ui.model.SwipeCard
import com.appsamurai.storyly.storyly_placement_flutter.common.data.product.encodeSTRProductItem

fun encodeSwipeCardDataPayload(data: SwipeCardDataPayload): Map<String, Any?> {
    return mapOf(
        "type" to STRWidgetType.SwipeCard.raw,
        "items" to data.items.map { encodeSTRProductItem(it) }
    )
}

fun encodeSwipeCardPayload(payload: SwipeCardPayload?): Map<String, Any?>? {
    payload ?: return null
    return mapOf(
        "card" to encodeSwipeCard(payload.card),
    )
}

fun encodeSwipeCard(card: SwipeCard?): Map<String, Any?>? {
    card ?: return null
    return mapOf(
        "actionProducts" to card.actionProducts?.map { encodeSTRProductItem(it) }
    )
}

File: placement-reactnative/storyly-placement-react-native/ios/Common/Data/Widgets/SwipeCardData.swift
import Foundation
import StorylyCore
import StorylySwipeCard

// MARK: - SwipeCard Data Payload Encoding

func encodeSwipeCardDataPayload(_ data: SwipeCardDataPayload) -> [String: Any] {
    return [
        "type": STRWidgetType.swipeCard.description,
        "items": data.items.map { encodeSTRProductItem($0) }
    ]
}

func encodeSwipeCardPayload(_ payload: SwipeCardPayload?) -> [String: Any]? {
    guard let payload = payload else { return nil }
    
    return [
        "card": encodeSwipeCard(payload.card)
    ].compactMapValues { $0 }
}

// MARK: - SwipeCard Encoding

func encodeSwipeCard(_ card: SwipeCard?) -> [String: Any]? {
    guard let card = card else { return nil }
    
    return [
        "actionProducts": card.actionProducts?.map { encodeSTRProductItem($0) }
    ].compactMapValues { $0 }
}


